terminus/models/road.py

```diff
@@ -91,6 +92,33 @@ def reverse(self):
     def be_two_way(self):
         pass
 
+    def length(self, initial=0, final=None):
+        distances = self.get_waypoint_distances()
+        if final is None:
+            return math.fsum(distances[initial:])
+        return math.fsum(distances[initial:final])
+
+    def get_waypoint_positions(self):
+        waypoints = self.get_waypoints()
```

Comment:
check the `map` function in Python. You can write this just as `map(lambda waypoint: waypoint.center, self.get_waypoints())`